addons/crm_survey/models/survey_user_input.py

```diff
@@ -0,0 +1,162 @@
+                        if question_id.question_type == "char_box":
+                            if not user_nickname and question_id.save_as_nickname:
+                                user_nickname = input_line._get_answer_value()
+                            if not public_user_mail and question_id.validation_email:
+                                public_user_mail = input_line._get_answer_value()
+
+                        # Write description line
+                        if question_id.question_type == "text_box":
+                            answer_description = Markup("<br/>&emsp;%s" % escape(input_line._get_answer_value()).replace('\n', Markup("<br/>&emsp;")))
+                        elif question_id.question_type == "matrix":
+                            row_id = input_line.matrix_row_id
+                            if row_id.display_name:
+                                col_value = input_line.suggested_answer_id.display_name
+                                if last_row_id != row_id:
+                                    last_row_id = row_id
+                                    answer_description += Markup("<br/>&emsp;%s — %s" % (escape(row_id.display_name), escape(col_value)))
+                                else:  # For multiple suggested answers
+                                    answer_description += Markup(", %s" % escape(col_value))
+                            else:  # Comment field
+                                comment_answer = input_line._get_answer_value()
+                                answer_description += Markup("<br/>&emsp;<i><b>%s</b></i> — %s" % (_('Comment'), escape(comment_answer).replace('\n', Markup("<br/>&emsp;"))))
+                        else:
+                            answer = escape(str(input_line._get_answer_value())).replace('\n', Markup("<br/>&emsp;"))
+                            if question_id.question_type in ["simple_choice", "multiple_choice"] and input_line.answer_type == 'char_box':  # Comment case
+                                answer_description += Markup("<br/>&emsp;<i><b>%s</b></i> — %s" % (_('Comment'), answer))
+                            else:  # Normal case
+                                # For picture answers without label, we use the filename by the way
+                                if input_line.suggested_answer_id and not input_line._get_answer_value():
+                                    answer = escape(input_line.suggested_answer_id.value_image_filename)
+                                answer_description += Markup("%s%s" % ((" — " if input_line_index == 0 else ", "), answer))  # The "else" is useful for multiple suggested answers
+            else:
+                answer_description = Markup("<i> — %s </i>" % _('Skipped'))
+
+            questions_with_answers_description = Markup("<li>%s%s</li>" % (escape(question_id.title), answer_description))
+            html_input_lines.append(questions_with_answers_description)
```

Comment:
can inline, no need for the variable `questions_with_answers_description` actually